include/xtensor/xhistogram.hpp

```diff
@@ -426,6 +442,60 @@ namespace xt
                         xt::ones<typename std::decay_t<E1>::value_type>(data.shape()),
                         minlength);
     }
+
+    /**
+     * Get the number of items in each bin, given the fraction of items per bin.
+     * The output is such that the total number of items of all bins is exactly "N".
+     *
+     * @param N the number of items to distribute
+     * @param weights fraction of items per bin: a 1D container whose size is the number of bins
+     *
+     * @return 1D container with the number of items per bin
+     */
+    template <class E>
+    inline xt::xtensor<size_t,1> bin_items(size_t N, E&& weights)
+    {
+        if (weights.size() <= 1) {
```

Comment:
- formatting: opening curly brace should be on a new line
- warning: compare with `1u` or `size_t(1)`